Timeout when accessing TeamCity API

```
Unexpected exception occurred invoking async method: public void org.gradle.devprod.collector.teamcity.TeamcityExport.loadGbtTriggerBuilds()

 org.jetbrains.teamcity.rest.TeamCityConversationException: Failed to connect to https://builds.gradle.org/guestAuth/app/rest/builds/id:61484873: interrupted
 at org.jetbrains.teamcity.rest.TeamCityInstanceImpl$service$3.handleError(implementation.kt:156) ~[teamcity-rest-client-1.14.0.jar!/:na]
 at org.jetbrains.teamcity.rest.TeamCityInstanceImpl$service$3.handleError(implementation.kt:105) ~[teamcity-rest-client-1.14.0.jar!/:na]
 at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:242) ~[retrofit-1.9.0.jar!/:na]
 at com.sun.proxy.$Proxy94.build(Unknown Source) ~[na:na]
 at org.jetbrains.teamcity.rest.BuildImpl.fetchFullBean(implementation.kt:1204) ~[teamcity-rest-client-1.14.0.jar!/:na]
 at org.jetbrains.teamcity.rest.BuildImpl.fetchFullBean(implementation.kt:1200) ~[teamcity-rest-client-1.14.0.jar!/:na]
 at org.jetbrains.teamcity.rest.BaseImpl$fullBean$2.invoke(implementation.kt:650) ~[teamcity-rest-client-1.14.0.jar!/:na]
 at org.jetbrains.teamcity.rest.BaseImpl$fullBean$2.invoke(implementation.kt:629) ~[teamcity-rest-client-1.14.0.jar!/:na]
 at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74) ~[kotlin-stdlib-1.5.32.jar!/:1.5.32-release-578(1.5.32)]
 at org.jetbrains.teamcity.rest.BaseImpl.getFullBean(implementation.kt) ~[teamcity-rest-client-1.14.0.jar!/:na]
 at org.jetbrains.teamcity.rest.BuildImpl.getRevisions(implementation.kt:1310) ~[teamcity-rest-client-1.14.0.jar!/:na]
 at org.gradle.devprod.collector.teamcity.TeamCityBuildKt.toTeamCityBuild(TeamCityBuild.kt:28) ~[teamcity-export-0.0.1-SNAPSHOT.jar!/:na]
 at org.gradle.devprod.collector.teamcity.TeamcityClientService$loadTriggerBuilds$2$1.invoke(TeamcityClientService.kt:39) ~[teamcity-export-0.0.1-SNAPSHOT.jar!/:na]
 at org.gradle.devprod.collector.teamcity.TeamcityClientService$loadTriggerBuilds$2$1.invoke(TeamcityClientService.kt:19) ~[teamcity-export-0.0.1-SNAPSHOT.jar!/:na]
 at kotlin.sequences.TransformingSequence$iterator$1.next(Sequences.kt:210) ~[kotlin-stdlib-1.5.32.jar!/:1.5.32-release-578(1.5.32)]
 at kotlin.sequences.FilteringSequence$iterator$1.calcNext(Sequences.kt:170) ~[kotlin-stdlib-1.5.32.jar!/:1.5.32-release-578(1.5.32)]
 at kotlin.sequences.FilteringSequence$iterator$1.hasNext(Sequences.kt:194) ~[kotlin-stdlib-1.5.32.jar!/:1.5.32-release-578(1.5.32)]
 at kotlin.sequences.FlatteningSequence$iterator$1.ensureItemIterator(Sequences.kt:307) ~[kotlin-stdlib-1.5.32.jar!/:1.5.32-release-578(1.5.32)]
 at kotlin.sequences.FlatteningSequence$iterator$1.hasNext(Sequences.kt:303) ~[kotlin-stdlib-1.5.32.jar!/:1.5.32-release-578(1.5.32)]
 at org.gradle.devprod.collector.teamcity.TeamcityExport.loadGbtTriggerBuilds(TeamcityExport.kt:62) ~[teamcity-export-0.0.1-SNAPSHOT.jar!/:na]
 at org.gradle.devprod.collector.teamcity.TeamcityExport$$FastClassBySpringCGLIB$$4dccbea2.invoke(<generated>) ~[teamcity-export-0.0.1-SNAPSHOT.jar!/:na]
 at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.18.jar!/:5.3.18]
 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783) ~[spring-aop-5.3.18.jar!/:5.3.18]
 at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.18.jar!/:5.3.18]
 at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753) ~[spring-aop-5.3.18.jar!/:5.3.18]
 at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115) ~[spring-aop-5.3.18.jar!/:5.3.18]
 at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
 at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
 at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
 at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]
 Caused by: retrofit.RetrofitError: interrupted
 at retrofit.RetrofitError.networkError(RetrofitError.java:27) ~[retrofit-1.9.0.jar!/:na]
 at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:395) ~[retrofit-1.9.0.jar!/:na]
 at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240) ~[retrofit-1.9.0.jar!/:na]
 ... 27 common frames omitted
 Caused by: java.io.InterruptedIOException: interrupted
 at okio.Timeout.throwIfReached(Timeout.kt:98) ~[okio-jvm-2.8.0.jar!/:na]
 at okio.InputStreamSource.read(JvmOkio.kt:87) ~[okio-jvm-2.8.0.jar!/:na]
 at okio.AsyncTimeout$source$1.read(AsyncTimeout.kt:129) ~[okio-jvm-2.8.0.jar!/:na]
 at okio.RealBufferedSource.request(RealBufferedSource.kt:206) ~[okio-jvm-2.8.0.jar!/:na]
 at okio.RealBufferedSource.require(RealBufferedSource.kt:199) ~[okio-jvm-2.8.0.jar!/:na]
 at okio.RealBufferedSource.readHexadecimalUnsignedLong(RealBufferedSource.kt:381) ~[okio-jvm-2.8.0.jar!/:na]
 at okhttp3.internal.http1.Http1ExchangeCodec$ChunkedSource.readChunkSize(Http1ExchangeCodec.kt:429) ~[okhttp-4.9.3.jar!/:na]
 at okhttp3.internal.http1.Http1ExchangeCodec$ChunkedSource.read(Http1ExchangeCodec.kt:408) ~[okhttp-4.9.3.jar!/:na]
 at okhttp3.internal.connection.Exchange$ResponseBodySource.read(Exchange.kt:276) ~[okhttp-4.9.3.jar!/:na]
 at okio.RealBufferedSource.read(RealBufferedSource.kt:189) ~[okio-jvm-2.8.0.jar!/:na]
 at okio.RealBufferedSource.exhausted(RealBufferedSource.kt:197) ~[okio-jvm-2.8.0.jar!/:na]
 at okio.GzipSource.read(GzipSource.kt:88) ~[okio-jvm-2.8.0.jar!/:na]
 at okio.RealBufferedSource$inputStream$1.read(RealBufferedSource.kt:158) ~[okio-jvm-2.8.0.jar!/:na]
 at retrofit.ExceptionCatchingTypedInput$ExceptionCatchingInputStream.read(ExceptionCatchingTypedInput.java:64) ~[retrofit-1.9.0.jar!/:na]
 at java.base/sun.nio.cs.StreamDecoder.readBytes(StreamDecoder.java:284) ~[na:na]
 at java.base/sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:326) ~[na:na]
 at java.base/sun.nio.cs.StreamDecoder.read(StreamDecoder.java:178) ~[na:na]
 at java.base/java.io.InputStreamReader.read(InputStreamReader.java:181) ~[na:na]
 at com.google.gson.stream.JsonReader.fillBuffer(JsonReader.java:1290) ~[gson-2.9.0.jar!/:na]
 at com.google.gson.stream.JsonReader.nextNonWhitespace(JsonReader.java:1328) ~[gson-2.9.0.jar!/:na]
 at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:538) ~[gson-2.9.0.jar!/:na]
 at com.google.gson.stream.JsonReader.peek(JsonReader.java:425) ~[gson-2.9.0.jar!/:na]
 at com.google.gson.Gson.assertFullConsumption(Gson.java:963) ~[gson-2.9.0.jar!/:na]
 at com.google.gson.Gson.fromJson(Gson.java:957) ~[gson-2.9.0.jar!/:na]
 at retrofit.converter.GsonConverter.fromBody(GsonConverter.java:63) ~[retrofit-1.9.0.jar!/:na]
 at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:367) ~[retrofit-1.9.0.jar!/:na]
 ... 28 common frames omitted
```

--- a/teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt
+++ b/teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt
@@ -14,14 +14,16 @@ import reactor.core.publisher.Mono
 import java.net.URI
 import java.time.Instant
 import java.time.temporal.ChronoUnit
+import java.util.concurrent.TimeUnit
 
 @Service
 class TeamcityClientService(
     @Value("${'$'}{teamcity.api.token}") private val teamCityApiToken: String,
     private val objectMapper: ObjectMapper
 ) {
-    private val teamCityInstance: TeamCityInstance =
-        TeamCityInstanceFactory.guestAuth("https://builds.gradle.org")
+    private val teamCityInstance: TeamCityInstance = TeamCityInstanceFactory
+        .guestAuth("https://builds.gradle.org")
+        .withTimeout(5, TimeUnit.MINUTES)
 
     private val client: WebClient = WebClient.create()
 

